protob/messages.proto

```diff
@@ -1437,3 +1452,113 @@ message DebugLinkMemoryWrite {
 message DebugLinkFlashErase {
 	optional uint32 sector = 1;
 }
+
+/**
+ * Request: Ask device for Cardano address
+ * @next CardanoAddress
+ * @next Failure
+ */
+ message CardanoGetAddress {
+	repeated uint32 address_n = 1;		// BIP-32-style path to derive the key from master node
+	optional bool show_display = 2;		// optionally prompt for confirmation on trezor display
+}
+
+/**
+ * Request: Ask device for Cardano address
+ * @next CardanoAddress
+ * @next Failure
+ */
+ message CardanoAddress {
+	optional string address = 1;		// Base58 cardano address
+}
+
+/**
+ * Request: Ask device to sign Cardano message
+ * @next CardanoMessageSignature
+ * @next Failure
+ */
+message CardanoSignMessage {
+	repeated uint32 address_n = 1;					// BIP-32 path to derive the key from master node
+	optional bytes message = 2;					// message to be signed
+}
+
+/**
+ * Request: Ask device for public key corresponding to address_n path
+ * @next CardanoPublicKey
+ * @next Failure
+ */
+ message CardanoGetPublicKey {
+	repeated uint32 address_n = 1;				// BIP-32 path to derive the key from master node
+}
+
+/**
+ * Response: Contains public key derived from device private seed
+ * @prev CardanoGetPublicKey
+ */
+ message CardanoPublicKey {
+	optional string xpub = 1;				// Xpub key
+	optional HDNodeType node = 2;				// BIP-32 public node
+	optional string root_hd_passphrase = 3;			// Hd passphrase for root in hex format
+}
+
+/**
+ * Request: Ask device to verify Cardano message
+ * @next Success
+ * @next Failure
+ */
+message CardanoVerifyMessage {
+	optional bytes public_key = 1;				// Public key which was used to sign message
```

Comment:
For other coins we use `address` not `public_key` in `VerifyMessage`, would it make sense to use it here too? (You don't return `public_key` but `address` in `CardanoMessageSignature` so it makes sense)

Reply:
Ah, correct. Let's use `optional bytes public_key = 1;` in `CardanoMessageSignature` for consistency.

After this change, the messages `CardanoSignMessage`, `CardanoVerifyMessage` and `CardanoMessageSignature` will be exactly the same as `LiskSignMessage`, `LiskVerifyMessage` and `LiskMessageSignature`, which is another Ed25519 currency, so it makes sense

Reply:
fixed